gemini/agent-engine/tutorial_ag2_on_agent_engine.ipynb

```diff
@@ -0,0 +1,930 @@
+        "        )\n",
+        "\n",
+        "\n",
+        "print(f\"Research App with multi-agents created.\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9Kwmo5QJx8Dv"
+      },
+      "source": [
+        "### Test\n",
+        "\n",
+        "This section demonstrates how to test the Agent Engine locally before deploying it to Vertex AI. By running the `ResearchApp` locally, you can observe a multi-agent conversation flow similar to the previous example. This local testing helps ensure that the agents and their interactions are functioning as expected before deploying to the cloud."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "z89kARQryAfb"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Testing - Local Development\n",
+        "local_engine = ResearchApp(project_id=PROJECT_ID, location=LOCATION)\n",
+        "local_engine.set_up()\n",
+        "report = local_engine.query(topic=research_topic)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "cellView": "form",
+        "id": "uM5QZ77lfo0Q"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Inspect Report\n",
+        "display(Markdown(report))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eOk6mEfYxl9n"
+      },
+      "source": [
+        "### Setup and Deploy to Vertex AI Agent Engine\n",
+        "\n",
+        "This section guides you through deploying the AG2 multi-agent application to Vertex AI Agent Engine. Before proceeding with the deployment, we'll ensure there are no existing deployments to avoid conflicts. This involves listing and deleting any previous Agent Engine deployments to start with a clean slate. Subsequently, we'll utilize the `AgentEngine.create` method to deploy the application, specifying dependencies and configurations.\n",
+        "\n",
+        "> **NOTE:** Please be patient as the deployment is a long running operation. Once successful, the cell will return the resource URI of deployment that can be used for subsequent executions.\n",
+        "> \n",
+        "> **Example:**\n",
+        "> ```py\n",
+        "> agent_engine = vertexai.agent_engines.get('projects/<YOUR_PROJECT_ID_NUMBER>/locations/us-central1/reasoningEngines/<DEPLOYED_AGENT_ENGINE_ID>')\n",
+        "> ```"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "cellView": "form",
+        "id": "BpXwgjnrXLto"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Clean up deployment if needed\n",
+        "\n",
+        "\n",
+        "def cleanup_deployments(dry_run: bool = True):\n",
+        "    agent_engine_list = agent_engines.AgentEngine.list()\n",
```

Comment:
Do you need to use `agent_engines.AgentEngine`? You already import the `AgentEngine` class on its own